### PR TITLE
[8.0] Add testing cases to stock modules

### PR DESCRIPTION
**Pull request by osvalr**
# Changes made:
- Unlink cost lines using `unlink()` when `onchange_invoice_ids` is triggered
- [x] rebase

### CODE REVIEW

File: stock_landed_costs_average/model/stock_landed_costs.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ class StockLandedCost(models.Model):
     @api.onchange('invoice_ids')
     def onchange_invoice_ids(self):
         for lc_brw in self:
-            lc_brw.update({'cost_lines': [(6, False, {})]})
+            lc_brw.cost_lines.unlink()
```
Comment (hbto) [inline stock_landed_costs_average/model/stock_landed_costs.py:36]: @osvalr I have to check the use of this unlink here,
I had some problems before that is the reason of using update with (6, False {})

File: stock_landed_costs_average/model/stock_landed_costs.py [hunk 2]
```diff
@@ -60,7 +60,6 @@ def onchange_invoice_ids(self):
                         'price_unit': price_subtotal,
                         'split_method': 'by_quantity',
                     }))
-            lc_brw.update({'cost_lines': [(6, False, {})]})
```
Comment (hbto) [inline stock_landed_costs_average/model/stock_landed_costs.py:63]: @osvalr I agree on taking away duplicated code

File: stock_landed_costs_average/model/stock_landed_costs.py (hunk 1 above)
Comment (osvalr) [inline stock_landed_costs_average/model/stock_landed_costs.py:36]: ok, let me know then

**Comment by hbto** [comment]
@osvalr this test keeps failing

openerp.addons.product_extended_segmentation.tests.test_avg_costs: ` AssertionError: 35.0 == 35.0

File: product_extended_segmentation/tests/test_avg_costs.py [hunk 3]
```diff
@@ -112,3 +112,15 @@ def test_02_compute_price_with_real_bom(self):
...
                          '{{{0}: 75.0}}'.format(str(template_id.id)))
+
+    def test_03_write_real_cost_product_price_using_wizard(self):
+        template_id = self.prod_e_id.product_tmpl_id
+        self.prod_d_id.write({
+            'cost_method': 'real',
+            'standard_price': 35.0
+        })
+        old_price = self.prod_d_id.standard_price
+        self.env['product.template'].compute_price(
```
Comment (hbto) [inline product_extended_segmentation/tests/test_avg_costs.py:123]: @osvalr as your initial value is higher than the new value to be written into standard_price
method is not writing anything, We were bitten by the dog.
Do you remember threshold? XD

Value that is being tried to write is 35.0,
We have to set a lower standard_price value at line 120
let us say 20.0

**Comment by hbto** [comment]
@osvalr I have fixed it
Waiting for CI

**Pull request merged**